samcli/lib/build/build_graph.py

```diff
@@ -115,54 +187,122 @@ def _read(self):
+                                                                         layer_build_definition_key])
+            self._layer_build_definitions.append(layer_build_definition)
+
+        # return self._build_definitions
 
     def _write(self):
         """
         Writes build definition details into build.toml file, which would be used by the next build.
         build.toml file will contain the same information as build graph,
         function details will only be preserved as function names
+        layer details will only be preserved as layer names
         """
         # convert build definition list into toml table
-        build_definitions_table = tomlkit.table()
-        for build_definition in self._build_definitions:
-            build_definition_as_table = _build_definition_to_toml_table(build_definition)
-            build_definitions_table.add(build_definition.uuid, build_definition_as_table)
+        function_build_definitions_table = tomlkit.table()
+        for build_definition in self._function_build_definitions:
+            build_definition_as_table = _function_build_definition_to_toml_table(build_definition)
+            function_build_definitions_table.add(build_definition.uuid, build_definition_as_table)
+
+        layer_build_definitions_table = tomlkit.table()
+        for build_definition in self._layer_build_definitions:
+            build_definition_as_table = _layer_build_definition_to_toml_table(build_definition)
+            layer_build_definitions_table.add(build_definition.uuid, build_definition_as_table)
 
         # create toml document and add build definitions
         document = tomlkit.document()
         document.add(tomlkit.comment("This file is auto generated by SAM CLI build command"))
-        document.add(BuildGraph.BUILD_DEFINITIONS, build_definitions_table)
+        document.add(BuildGraph.FUNCTION_BUILD_DEFINITIONS, function_build_definitions_table)
+        document.add(BuildGraph.LAYER_BUILD_DEFINITIONS, layer_build_definitions_table)
 
         if not self._filepath.exists():
             open(self._filepath, "a+").close()
 
         self._filepath.write_text(tomlkit.dumps(document))
 
 
-class BuildDefinition:
+class AbstractBuildDefinition:
     """
+    Abstract class for build definition
     Build definition holds information about each unique build
     """
 
-    def __init__(self, runtime, codeuri, metadata):
+    def __init__(self, source_md5):
+        self.uuid = str(uuid4())
+        self.source_md5 = source_md5
+
+    def set_source_md5(self, new_source_md5):
+        self.source_md5 = new_source_md5
+
+    def get_source_md5(self):
+        return self.source_md5
+
+    def get_uuid(self):
+        return self.uuid
+
+
+class LayerBuildDefinition(AbstractBuildDefinition):
+    """
+    LayerBuildDefinition holds information about each unique layer build
+    """
+    def __init__(self, name, codeuri, build_method, compatible_runtimes, source_md5=''):
```

Comment:
nit: single quotes; was this ran through `black`?

Reply:
Reformat it to double quotes.